src/logic/safe/store/reducer/pendingTransactions.ts

```diff
@@ -46,7 +46,7 @@ export const pendingTransactionsReducer = handleActions<PendingTransactionsState
       // Omit id from the pending transactions on current chain
       const { [id]: _, ...newChainState } = state[chainId] || {}
 
-      if (Object.keys(newChainState[chainId] || {}).length === 0) {
+      if (Object.keys(newChainState || {}).length === 0) {
```

Comment:
Is this a general bugfix that should be in main?

Reply:
It was wiping out a `[chainId]` pendingTxs per complete when entering this switch case.

Reply:
We should clear any remaining 'empty' pending `chainId`s but I see the error here. It falls back to an empty object. I'll push a fix for it now.

Reply:
Here's the fix: https://github.com/gnosis/safe-react/pull/3685

Reply:
This logic is required by the pending watcher. However, I can see that this clause will always be true. I've pushed a fix here: https://github.com/gnosis/safe-react/pull/3688